test_suite/conftest.py

```diff
@@ -32,15 +39,36 @@ def check_markers(host, request):
     ]
 
     host_distro = host.system_info.distribution
-    host_version = host.system_info.release
+    host_version = float(host.system_info.release)
     distro_version = f'{host_distro}{host_version}'
 
     # Skip the test if the distro is explicitly excluded
     if exclude_on_marker:
         exclude_on_marker_list = exclude_on_marker.args[0]
         if host_distro in exclude_on_marker_list \
                 or distro_version in exclude_on_marker_list:
-            pytest.skip(f"This test doesn't apply to {distro_version}")
+            pytest.skip(skip_massage.format(distro_version=distro_version))
+
+        elif len(exclude_on_marker_list) > 1 and host_distro in exclude_on_marker_list[0]:
+            exclude_on_marker_list_version = parse_distro_version(exclude_on_marker_list[0])
+            if exclude_on_marker_list[1] == '<':
+                if host_version < exclude_on_marker_list_version:
+                    pytest.skip(f"Skip test with version smaller than {exclude_on_marker_list[0]}")
+            elif exclude_on_marker_list[1] == '>':
+                if host_version > exclude_on_marker_list_version:
+                    pytest.skip(f"Skip test with version bigger than {exclude_on_marker_list[0]}")
+
+    if major_version_marker:
+        major_version_marker_list = major_version_marker.args[0]
+        for major_version in major_version_marker_list:
+            major_version_marker_list_version = parse_distro_version(major_version)
+            if host_version >= major_version_marker_list_version:
```

Comment:
I think this first "if" is redundant since the second one will do the check we need.

Reply:
The first if checks if the version is bigger the the "major version" we compare, since the versions are float we need to make so that this scenario for example will succeed as well:
major version = 8.5
host version = 9.0
In this case we want to skip this test for host version with 9.0 because this is not as part of the major release, we should run only tests 8.5,8.6,8.7,8.8,8.9
The second skip is necessary for this scenario

In addition, if we leave only the second if we will fail also in this scenario:
major version = 8.5
host version = 8.3
It will not skip the version 8.3 even though it should because both int are 8

Reply:
Then the name of the marker is misleading or it has more than one purpose. If it's called "major_version" I would think or assume that it is only for filtering by major versions here (I bet that other contributors may reach to the same conclusion at first sight). Why would this marker need to check for minor versions as well? I think having a separate marker for this serves also for the purpose of separating the logic that is only used for major version checks.